File: InterIF/InterIF-2018/H.py
N = int(input())
X = int(input())

pessoas = [i for i in range(1,N+1)]
mortes = 0
i = 1
while(len(pessoas) > 1):
    mortes+=1
    if(mortes == X):
        print(f'Morte {mortes}: {pessoas[i]}')
    del pessoas[i]
    if(i > len(pessoas) - 1):
        i = 1
    elif(i + 1 > len(pessoas) -1 ):
        i = 0
    else:
        i+=1

print(f'Sobrevivente: {pessoas[0]}')

File: InterIF/InterIF-2018/H.cpp
#include <iostream>
#include <vector>

using namespace std;

void optimize()
{
    // ios_base::sync_with_stdio(0);
    // cin.tie(false);
}

int main()
{

    optimize();

    long N, X, temp;
    long mortes = 0;
    long i = 0;
    vector<long> pessoas;
    cin >> N;
    cin >> X;

    while (pessoas.size() > 1)
    {
        mortes++;
        if (mortes == X)
        {
            cout << "Morte " << mortes << ": " << pessoas[i] << "\n";
        }
        pessoas.erase(pessoas.begin() + i);
        if (i > pessoas.size() - 1)
        {
            i = 1;
        }
        else if (i + 1 > pessoas.size() - 1)
        {
            i = 0;
        }
        else
        {
            i++;
        }
    }

    cout << "Sobrevivente: " << pessoas[0] << "\n";

    return 0;
}